routemaster/jobs/batch.rb

```diff
@@ -0,0 +1,57 @@
+require 'routemaster/jobs'
+require 'routemaster/models/batch'
+require 'routemaster/services/deliver'
+require 'routemaster/mixins/log'
+
+module Routemaster
+  module Jobs
+    class Batch
+      include Mixins::Log
+
+      def initialize(delivery: Services::Deliver)
+        @delivery = delivery
+      end
+
+      def call(uid)
+        batch = Models::Batch.new(uid: uid)
+
+        # prevents further ingestion in this batch (idempotent)
+        batch.promote
+
+        subscriber = batch.subscriber
+        data = batch.data
+
+        # handle unsubscription, autodrop
+        if subscriber.nil? || data.nil? || data.empty?
+          batch.delete
+          return self
+        end
```

Comment:
This feels like it should be a method on `Models::Batch`, e.g. `deliverable?` or something along those lines which would make this (rather long) method shorter, and more testable.